pkg/extension/sumologicextension/extension.go

```diff
@@ -166,6 +170,11 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
 		zap.String(collectorIdField, colCreds.Credentials.CollectorId),
 	)
 
+	err = se.updateMetadataWithHTTPClient(ctx, se.httpClient)
+	if err != nil {
+		return err
```

Comment:
Should failure to update metadata result in an error? Should we only log it?

Reply:
Should we make further attempts (with backoff)? How important do we consider the metadata?

Reply:
metaData has many sub parts, like agent version, hostDetails and tags.
And it is used to populate the Agent List page in UI. Without these details, List page is not useful.
And updating tags has also other use cases , like tag will be used later for source, subscription and searching/sorting.
So, We should make attempts with backoff in case of failure.
On server side, we can also plan to generate a low priority alert if metadata update fails for more than x times for a customer.

Reply:
yes, failure to update metadata must be an error

Reply:
Hm, if this results in an error it might be a major pain for customers.

As long as we can send the data to Sumo I believe we should, even if we have degraded funcionality because collectors metadata are not there.

I believe that telemetry data are more important than collectors metadata.

Reply:
I am going to update this PR to implement retry w/ backoff and log any/all related errors ("loudly" 😅).